.github/workflows/run-tests.yml

```
@@ -7,7 +7,13 @@ on:
     types: [completed]
 
 jobs:
-  run:
+  on-failure:
+    name: Notify on failure
+    runs-on: ${{ matrix.os }}
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
+      - run: echo 'The triggering workflow failed.'
```

```suggestion
    steps:
      - run: |
           echo 'The triggering workflow failed.'
           0
```

You'll also want to return status 0 to the CI in order for it to not deliver a false positive (check mark as 'passed' if it didn't even run)

Example: https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/workflows/deploy-tests.yml#L25-L27

---

Though the indenting in my suggestion may need to be corrected (two spaces from the start of the 'run' string)